### PR TITLE
WIP [WEB-1371] Automate Debranding Styles

### PR DESCRIPTION
**Pull request by jeffkamo**
**Jira Ticket:** [WEB-1371](https://mobify.atlassian.net/browse/WEB-1371)
**Related PR:** [platform-scaffold#833](https://github.com/mobify/platform-scaffold/pull/833), [platform-generater#42](https://github.com/mobify/platform-generator/pull/42)

## Changes:
- Add a new command to the generator script that replaces CSS property values with its corresponding `ui_kit_value` declaration.

## Todos
- [x] Set the `SCAFFOLD_VERSION_OR_BRANCH` value back to `master`

## Post Merge Todos:
- [x] Delete the [temporary release](https://github.com/mobify/platform-scaffold/releases/tag/temp-debrand-automation) that was used for testing purposes

## Future Considerations
- Add debranding automation to remove or tweak branded colors in image assets and markup files (`.html` and `.jsx` files)

## How to Test
1. Run the following command:
    ```bash
    bash <(curl -fsS https://raw.githubusercontent.com/mobify/generator-progressive-web/WEB-1371_automate-debranding-styles/generate.sh)
    ```
2. Follow the generator instructions
3. Wait for the `npm install` (which starts automatically by the generator) to finish
4. Change into the newly generated project's web directly, i.e. `cd [project-slug-name]/web`
5. Run `npm run dev`
6. Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
7. **Verify** that the design remains intact, but Merlin's branded color values have been swapped with UI Kit color values. (You can download the latest [UI Kit Sketch file here](https://github.com/mobify/progressive-mobile-ui-kit/releases/tag/v2.2.0))
8. **Verify** that no `ui_kit_value` statements are left in the generated project

> _**Note** that some colors can't be automatically removed, such as those in image assets, SVG or JSX files (i.e. those added from utility classes, etc.)_

## Result

| Before | After |
| --- | --- |
| before | screen shot 2017-06-28 at 2 32 00 pm |

### CODE REVIEW

File: generate.sh [hunk 1]
```diff
@@ -2,7 +2,8 @@
 set -o pipefail
 
 MYDIR=$(pwd)
-SCAFFOLD_VERSION_OR_BRANCH="master"
+# SCAFFOLD_VERSION_OR_BRANCH="master"
+SCAFFOLD_VERSION_OR_BRANCH="temp-debrand-automation"
```
Comment (jeffkamo) [inline generate.sh]: This will return to normal BEFORE merging. It's here for testing purposes.

**Comment by kevinxh** [comment]
I tried to generate a project with `slug: merlins-potions-2`..

but hanging on loading state:
screen shot 2017-07-05 at 2 01 37 pm

During project generation, I noticed `sed 1: "s/http:...." bag flag in substitute command`, wonder if it has anything to do with it...
screen shot 2017-07-05 at 1 51 31 pm

**Comment by kevinxh** [comment]
I noticed that the social media icon background still have Merlins brand color, wondering if that's something missing?

**Update:** Nevermind, I thought they are SDK components, in fact they are just static svgs.

screen shot 2017-07-05 at 2 57 12 pm

File: generate.sh [hunk 2]
```diff
@@ -61,6 +62,9 @@ sed -i -e "s/Merlin's/$project_name/g" web/app/static/manifest.json
 # This is about the web
 cd web || exit
 
+# Replace property values across stylesheets with their respective ui_kit_value
```
Comment (benlast) [inline generate.sh:64]: This isn't limited to stylesheet files - I don't know if that's a problem? If so, here's one way to do that:
```bash
egrep -lR "ui_kit_value" $(find ./app -name '*.css' -o -name '*.scss') | xargs ...
```
(you might need to edit the list of possible file extensions in the `find`. `-o` means _or_)

**Review by benlast — COMMENTED** [review]
I didn't even know this generator existed until a couple of days ago... it doesn't do any Messaging setup, so anything that uses it will not have functioning Messaging. Anyway, I added a comment on the way the processing is done.

**Comment by jeffkamo** [comment]
@benlast ya, I think that is pretty unlikely. I don't foresee us moving away from SCSS (or CSS) anytime soon.

**Comment by jeffkamo** [comment]
got verbal +1 from Ben. Merging!

**Pull request merged**